File: templates/problems/01_arrays_hashing/01_contains_duplicate/go/contains_duplicate_test.go
package contains_duplicate

import "testing"

func TestContainsDuplicate(t *testing.T) {
	tests := []struct {
		name string
		nums []int
		want bool
	}{
		{
			name: "has duplicates",
			nums: []int{1, 2, 3, 1},
			want: true,
		},
		{
			name: "no duplicates",
			nums: []int{1, 2, 3, 4},
			want: false,
		},
		{
			name: "empty slice",
			nums: []int{},
			want: false,
		},
		{
			name: "single element",
			nums: []int{1},
			want: false,
		},
		{
			name: "all same elements",
			nums: []int{5, 5, 5, 5},
			want: true,
		},
		{
			name: "duplicates at end",
			nums: []int{1, 2, 3, 4, 5, 5},
			want: true,
		},
		{
			name: "negative numbers with duplicates",
			nums: []int{-1, -2, -3, -1},
			want: true,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := containsDuplicate(tt.nums)
			if got != tt.want {
				t.Errorf("containsDuplicate(%v) = %v, want %v", tt.nums, got, tt.want)
			}
		})
	}
}

File: templates/problems/01_arrays_hashing/01_contains_duplicate/typescript/containsDuplicate.test.ts
import { describe, it, expect } from 'vitest';
import { containsDuplicate } from './containsDuplicate';

describe('ContainsDuplicate', () => {
    it('has duplicates', () => {
        expect(containsDuplicate([1, 2, 3, 1])).toBe(true);
    });

    it('no duplicates', () => {
        expect(containsDuplicate([1, 2, 3, 4])).toBe(false);
    });

    it('empty slice', () => {
        expect(containsDuplicate([])).toBe(false);
    });

    it('single element', () => {
        expect(containsDuplicate([1])).toBe(false);
    });

    it('all same elements', () => {
        expect(containsDuplicate([5, 5, 5, 5])).toBe(true);
    });

    it('duplicates at end', () => {
        expect(containsDuplicate([1, 2, 3, 4, 5, 5])).toBe(true);
    });

    it('negative numbers with duplicates', () => {
        expect(containsDuplicate([-1, -2, -3, -1])).toBe(true);
    });
});
